sys/mips/timer.c

```diff
@@ -27,9 +29,22 @@ static bintime_t mips_timer_gettime(timer_t *tm);
 static uint64_t read_count(mips_timer_state_t *state) {
   SCOPED_INTR_DISABLED();
   state->count.lo = mips32_get_c0(C0_COUNT);
+
   /* detect hardware counter overflow */
-  if (state->count.lo < state->last_count_lo)
+  if (state->count.lo < state->last_count_lo) {
+    state->cntr_modulo +=
+      ((uint32_t)(-1) - state->last_count_lo) + state->count.lo;
```

Comment:
If `last_count_lo` is 1 and `count.lo` is 0, then we want to increase `cntr_modulo` by `UINT32_MAX`, right? This code will increase it by `UINT32_MAX - 1`. I don't think we need to do different arithmetic in the two cases, I think `state->cntr_modulo += state->count.lo - state->last_count_lo` will work just fine in both cases.

Reply:
My bad I figured wrong, that between 1 and 0 you have `UINT32_MAX - 1`, thanks!